src/libp2p_swarm.erl

```diff
@@ -127,6 +128,21 @@ keys(TID) ->
     Server = libp2p_swarm_sup:server(TID),
     gen_server:call(Server, keys).
 
+-spec set_network_id(ets:tab() | pid(), binary()) -> ok.
+set_network_id(Sup, NetworkID) when is_pid(Sup) ->
+    set_network_id(tid(Sup), NetworkID);
+set_network_id(TID, NetworkID) ->
+    ets:insert(TID, {network_id, NetworkID}).
+
+-spec get_network_id(ets:tab() | pid()) -> binary() | undefined.
+get_network_id(Sup) when is_pid(Sup) ->
```

Comment:
Could you rename this to `network_id`? makes it consistent with other getters?

Reply:
Done